### PR TITLE
cmake: Fix build directory exclusion

### LINKED ISSUE #11443: [31.0.0-beta2] - source tarball is missing cmake/common/buildnumber.cmake
### Operating System Info

Other

### Other OS

Exherbo Linux

### OBS Studio Version

31.0.0-beta1

### OBS Studio Version (Other)

31.0.0-beta2

### OBS Studio Log URL

-

### OBS Studio Crash Log URL

_No response_

### Expected Behavior

Being able to run cmake without hitting any error.

### Current Behavior

```
CMake Error at cmake/common/bootstrap.cmake:61 (include):
  include could not find requested file:

    buildnumber
Call Stack (most recent call first):
  CMakeLists.txt:3 (include)
```

The source tarball https://github.com/obsproject/obs-studio/releases/download/31.0.0-beta2/OBS-Studio-31.0.0-beta2-Sources.tar.gz is missing the file(s)  https://github.com/obsproject/obs-studio/blob/31.0.0-beta2/cmake/common/buildnumber.cmake (and https://github.com/obsproject/obs-studio/blob/31.0.0-beta2/cmake/common/buildspec_common.cmake).

### Steps to Reproduce

1. cmake ../obs-studio-31.0.0-beta2-sources

### Anything else we should know?

_No response_

### PR DESCRIPTION
**Pull request by RytoEX**
### Description

The regex was incorrectly excluding any file with build in the name. The intent was to exclude any build directories, so we should be able to restrict this.

### Motivation and Context

Fixes #11443.

### How Has This Been Tested?

It hasn't.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [ ] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by PatTheMav** [comment]
I'm not sure if the "proper" fix is to remove the tarball entirely - given that only one user has reported that defect (and didn't even notice the entire extent of it), it seems like we already spent too much maintenance effort on a _very_ niche requirement.

**Comment by tgurr** [comment]
> I'm not sure if the "proper" fix is to remove the tarball entirely - given that only one user has reported that defect (and didn't even notice the entire extent of it), it seems like we already spent too much maintenance effort on a _very_ niche requirement.

@PatTheMav The tarball including the whole source is required as it includes submodules (like obs-browser/obs-websocket) which would be missing otherwise. The reason why you're not seeing others complaining is probably because no proper distribution actually bothers with trying to package beta versions, you'll likely see these pop up when a proper release is done with the things being broken. As I/we were hit by issues previously I got cautious in regards to obs and try to temporarily package beta versions as well so we can try to catch these issues like these early e.g. before a proper release has been done.

File: cmake/linux/cpackconfig.cmake [hunk 1]
```diff
@@ -16,7 +16,7 @@ endif()
 
 set(CPACK_SOURCE_PACKAGE_FILE_NAME "obs-studio-${CPACK_PACKAGE_VERSION}-sources")
 set(CPACK_SOURCE_GENERATOR "TGZ")
-set(CPACK_SOURCE_IGNORE_FILES "/.git" "/.*build.*" "/.ccache" "/.deps")
+set(CPACK_SOURCE_IGNORE_FILES "/.git" "/build.*" "/.ccache" "/.deps")
```
[the remark this answers is not in the document]
Comment (RytoEX) [reply cmake/linux/cpackconfig.cmake]: I've pushed a change that should fix this.

**Comment by RytoEX** [comment]
Pushed a change that should no longer exclude `/buildspec.json` or the other files. Do note that it will still exclude `/build-aux/`, but the current regex was already doing that.

**Comment by RytoEX** [comment]
After reviewing `.github/scripts/.package.zsh`, it seems like the only currently valid build directory on CI for the purpose of generating a source package via CPack is `build_x86_64`.
https://github.com/obsproject/obs-studio/blob/7979421cbf9ba2ece81ef1269b76c2ba022ab712/.github/scripts/.package.zsh#L51-L55
https://github.com/obsproject/obs-studio/blob/7979421cbf9ba2ece81ef1269b76c2ba022ab712/.github/scripts/.package.zsh#L239-L240

We could just do this instead:
```cmake
set(CPACK_SOURCE_IGNORE_FILES "/.git" "/build_x86_64" "/.ccache" "/.deps")
```

Perhaps it's not as elegant, but it would resolve the issue.

**Comment by PatTheMav** [comment]
> After reviewing `.github/scripts/.package.zsh`, it seems like the only currently valid build directory on CI for the purpose of generating a source package via CPack is `build_x86_64`.
> 
> https://github.com/obsproject/obs-studio/blob/7979421cbf9ba2ece81ef1269b76c2ba022ab712/.github/scripts/.package.zsh#L51-L55
> 
> 
> https://github.com/obsproject/obs-studio/blob/7979421cbf9ba2ece81ef1269b76c2ba022ab712/.github/scripts/.package.zsh#L239-L240
> 
> We could just do this instead:
> 
> ```cmake
> set(CPACK_SOURCE_IGNORE_FILES "/.git" "/build_x86_64" "/.ccache" "/.deps")
> ```
> 
> Perhaps it's not as elegant, but it would resolve the issue.

The `_x64_64` token is dynamic, _however_ you could probably use `CMAKE_SYSTEM_PROCESSOR` to match it, so `/build_${CMAKE_SYSTEM_PROCESSOR}` should give you the exact same name dynamically (and thus matches the code/behaviour used by the script to define the build directory).

**Comment by RytoEX** [comment]
> The `_x64_64` token is dynamic

In that specific part of the script, it isn't dynamic. The `.package.zsh` script bails out if the target is not in `_valid_targets`, so the part of the script that creates the sources package on the Ubuntu host can _only_ have `x86_64` as the arch at this time. This part of the script only runs on Ubuntu and not on macOS.

That said, `/build_${CMAKE_SYSTEM_PROCESSOR}` is probably fine, since on Ubuntu it _should_ return `x86_64`.

**Review by PatTheMav — APPROVED** [review]

**Pull request merged**